### ISSUE TITLE
New quest: Do these traffic lights have a button to request vibration on green light?

### ISSUE THREAD

**Issue by JensErat** [westnordost/StreetComplete#1330]
### General
Affected tag(s) to be modified/added: [`traffic_signals:vibration:yes/no`](https://wiki.openstreetmap.org/wiki/Key:traffic_signals:vibration)
Question asked: **Do these traffic lights have a button to request vibration on green light?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [X] 🚧 To be added tag is established and has a useful purpose
- [X] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [X] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [X] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [X] 🕓 Applies to a reasonable number of elements (Worth the effort)

### Ideas for implementation

**Element selection:** `traffic_lights`

**Metadata needed:** 

**Proposed GUI:** yes/no answer, or alternative to `traffic_lights:sound` (so far I never observed a traffic lights that have both sound and vibration, but might of course be possible)

At least in my home city, Stuttgart (Germany), most traffic lights have a button that activates vibrations on green signals. Only very few traffic lights have sound here. This is generally common in Germany.

**Comment by matkoniecz** [comment]
> Do these traffic lights have a button to request vibration on green light?

Note that this tag at https://wiki.openstreetmap.org/wiki/Key:traffic_signals:vibration is described as

> traffic_signals:vibration=yes/no is used with crossing=traffic_signals to indicate that vibrations occur when crossing is permitted. 

maybe something like

> Do these traffic lights have something vibrating on green light?

would be better. Thought "something" should be replaced by something more specific, is it always a button?

**Comment by JensErat** [comment]
I don't know any traffic lights vibrating _without_ a button, but that hardly counts as a proof. Although pressing a button for vibration seems to at least somehow makes sense -- this is mechanics and thus has more intensive tear than a speaker, and you have to touch something anyway.

Asking "Do vibrations occur when crossing is permitted?" does not really feel any better, either. At least it's shorter.

**Comment by PanderMusubi** [comment]
In the Netherlands you have none of these at the moment. So enabling this quest now would be useless for that country. Suppose these devices are implemented in a few years, they could not be added via this app anymore as for all the answer now would be no. How to deal best with these kind of quests?

**Comment by fishsunsun** [comment]
In my city, Hong Kong, some of the traffic light has vibration at the button of the activator. Also, sound is common (never seen one without) in Hong Kong.

**Comment by dbrgn** [comment]
What about changing the sound quest? So instead of asking "Does it make a sound? Yes/No" ask something like "Is there an audible or vibrating indication of a green light? Sound/Vibration/Both/No".

**Comment by PanderMusubi** [comment]
Sound is indeed much more common. Good to combine.

**Comment by rugk** [comment]
>  different vibrating patterns to indicate different pedestrian signal states

So, @fishsunsun, does this actually mean it also vibrates (just differently) when the light is red?
(I only know it so that it _only_ vibrates when the light is green.)

> something like "Is there an audible or vibrating indication of a green light? Sound/Vibration/Both/No".

The only issue could be available horizontal space… cough… https://github.com/westnordost/StreetComplete/issues/1284

**Comment by JensErat** [comment]
> In the Netherlands you have none of these at the moment. So enabling this quest now would be useless for that country. Suppose these devices are implemented in a few years, they could not be added via this app anymore as for all the answer now would be no. How to deal best with these kind of quests?

I guess you will never be able to solve this problem but for invalidating answers after some time and asking again. "Is this traffic lights still not having sound?" Same applies for pretty much anything else probably -- we're starting providing very detailed information here, and this will change all the time. I guess in European Union there has to be support for the visual impaired (and so on) as soon as some noteworthy construction is performed on roads, at least it looks like this.

> What about changing the sound quest? So instead of asking "Does it make a sound? Yes/No" ask something like "Is there an audible or vibrating indication of a green light? Sound/Vibration/Both/No".

Sounds reasonable to me.

**Comment by dbrgn** [comment]
> The only issue could be available horizontal space… cough… #1284 

This shouldn't be a problem, the street surface quest shows about 8-12 different options. Could be solved similarly.

**Comment by peternewman** [comment]
In the UK there's a "spinner" which operates in parallel with the green light (and sometimes instead of the audible indication) i.e. you don't need to explicitly request it by pressing a special button, just feel for it, see:
https://www.bbc.co.uk/news/blogs-ouch-22706881

**Comment by westnordost** [comment]
Because of the lack of information if they exist in other countries, let's first only enable it in countries where we know from commenters that it exists.

**Comment by peternewman** [comment]
Thanks for this @westnordost , can I ask a question and make a couple of observations.

If I've read the code correctly, you have to answer all three (vibration/noise/spinner) in one go yes?

If so, a couple of observations. Firstly I've often filled in the sound quest when not actually crossing the road e.g. if a crossing on the opposite side (e.g. a T junction), or while walking nearby I can hear the noise I complete that quest (or it might take a few attempts on a regular commute to complete a junction). This would no longer be possible with the current UI.

Also I think I'm right in saying this is the first request which requires you to touch anything (for the vibration bit), or in the UK look underneath, which is a challenge when you're above average height and touching would be easier. Is it really the best time during a global pandemic to introduce a quest which wants you to touch things when we have a virus which is known to live for a long time on hard surfaces (personally I'm trying my best to avoid touching pedestrian crossings, or elbow them if I have to). :mask: 

Finally, having just looked on the wiki, is it worth adding minimap too, I'd not heard of it (or seen it in the UK), or is not deemed used enough?

Would a UI like this not work with relatively minimal changes?:
Assistance|Yes|No
---|---|---
Sound| :o: | :red_circle: 
Arrow| :o: | :o: 
Vibration| :red_circle: | :o:
Mini Map| :o: | :o:
Submit|Merged|Row

Which in this case would only touch sound and vibration.

**Comment by westnordost** [comment]
> If I've read the code correctly, you have to answer all three (vibration/noise/spinner) in one go yes?

Yes.

Your proposed UI looks interesting. So it is intended for people who want to only answer it partially. However there is a problem with this. If you answered it partially, that would mean that the quest will not be solved. It will be shown to others again and right after upload, it will immediately be shown to you again because not all of sound, arrow and vibration (at least) are answered.

**Comment by smichel17** [comment]
@peternewman I think you should open a new issue for this. The idea of filling out multiple pieces of info at once has come up enough — ex: https://github.com/westnordost/StreetComplete/issues/102#issuecomment-689027242 and https://github.com/westnordost/StreetComplete/issues/1717#issuecomment-581188710 — that it would be nice to have one issue to refer back to about the concept in general. Especially now that there is a concrete UI suggestion.

**Comment by peternewman** [comment]
> @peternewman I think you should open a new issue for this.

Okay @smichel17 , I've done so in #2127.
